uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Document.php

```diff
@@ -12,6 +12,14 @@ public static function create(&$params)
     {
         $entityName = 'Document';
         $hook = empty($params['id']) ? 'create' : 'edit';
+
+        if ($hook === 'create') {
+            // If creating a new Document, we require all three contacts to be defined.
+            if (empty($params['source_contact_id'] || empty($params['target_contact_id']) || empty($params['assignee_contact_id']))) {
+                throw new CiviCRM_API3_Exception("Please specify 'source_contact_id', 'target_contact_id' and 'assignee_contact_id'.");
```

Comment:
Why throughing  CiviCRM_API3_Exception ... The API is not the only place that call this method ... as you know you can write :
`CRM_Tasksassignments_BAO_Document::create($params)` in any place in the so throwing API exception doesn't make much sense .
I think there is CiviCRM_Error_Exception or something like that could be used here.